xml_converter/src/packaging_protobin.cpp

```diff
@@ -284,3 +291,47 @@ void write_protobuf_file_per_map_id(
             &state);
     }
 }
+void _category_filter_by_depth(
+    const map<string, Category>* marker_categories,
+    int target_depth,
+    map<string, StringHierarchy>* category_hierarchy,
+    vector<string> parent_name,
+    int current_depth = 0) {
+    for (auto it = marker_categories->begin(); it != marker_categories->end(); it++) {
+        vector<string> child_name = parent_name;
+        child_name.push_back(it->first);
+        if (target_depth == current_depth) {
+            StringHierarchy category_filter;
+            category_filter.add_path(child_name, true);
+            (*category_hierarchy)[join(child_name, "_")] = category_filter;
+        }
+
+        _category_filter_by_depth(&(it->second.children), target_depth, category_hierarchy, child_name, current_depth + 1);
+    }
+}
+
+// Write protobuf per category
+void write_protobuf_file_per_category(
+    const string& marker_pack_root_directory,
+    const OptionalInt& split_by_category_depth,
```

Comment:
This should probably just be an int. There is no reason this function should be called with a non-int value, and your code is already assuming that this is an int by calling `.get_value()` without checking if it has a value or not.

Reply:
Good point. Since I check for a value before this function is called, it should always be an int.